packages/plugin-axe/src/lib/axe-plugin.ts

```diff
@@ -0,0 +1,50 @@
+import { createRequire } from 'node:module';
+import type { PluginConfig, PluginUrls } from '@code-pushup/models';
+import { normalizeUrlInput } from '@code-pushup/utils';
+import type { AxePluginOptions } from './config.js';
+import { AXE_DEFAULT_PRESET, AXE_PLUGIN_SLUG } from './constants.js';
+import { processAuditsAndGroups } from './processing.js';
+import { createRunnerFunction } from './runner/runner.js';
+
+/**
+ * Code PushUp plugin for accessibility testing using axe-core.
+ *
+ * @public
+ * @param urls - {@link PluginUrls} URL(s) to test
+ * @param options - {@link AxePluginOptions} Plugin options
+ * @returns Plugin configuration
+ */
+export function axePlugin(
+  urls: PluginUrls,
+  options?: AxePluginOptions,
+): PluginConfig {
+  const scoreTargets = options?.scoreTargets;
+  const preset = options?.preset ?? AXE_DEFAULT_PRESET;
```

Comment:
Users may not be using type-safe configs, so it's best to validate their inputs with the Zod schema first.

Zod could be used to fill in default values as well.